src/healthimaging-mcp-server/awslabs/healthimaging_mcp_server/healthimaging_operations.py

```diff
@@ -0,0 +1,713 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""AWS HealthImaging client for medical imaging operations."""
+
+# Standard library imports
+# Third-party imports
+import boto3
+import json
+import logging
+from botocore.exceptions import ClientError, NoCredentialsError
+from typing import Any, Dict, Optional
+
+
+logger = logging.getLogger(__name__)
+
+# HealthImaging API limits
+MAX_SEARCH_COUNT = 100  # Maximum number of resources per search request
+DATASTORE_ID_LENGTH = 32  # AWS HealthImaging datastore ID length
+
+
+def validate_datastore_id(datastore_id: str) -> str:
+    """Validate AWS HealthImaging datastore ID format."""
+    if not datastore_id or len(datastore_id) != DATASTORE_ID_LENGTH:
+        raise ValueError(f'Datastore ID must be {DATASTORE_ID_LENGTH} characters')
+    return datastore_id
+
+
+class HealthImagingSearchError(Exception):
+    """Exception raised for HealthImaging search parameter errors."""
+
+    def __init__(self, message: str, invalid_params: Optional[list[str]] = None):
+        """Initialize HealthImaging search error with message and optional invalid parameters."""
+        self.invalid_params = invalid_params or []
+        super().__init__(message)
+
+
+class HealthImagingClient:
+    """Client for AWS HealthImaging operations."""
+
+    def __init__(self, region_name: Optional[str] = None):
+        """Initialize the HealthImaging client."""
+        try:
+            self.session = boto3.Session()
```

Comment:
Please add the standardized user agent extra https://github.com/search?q=repo%3Aawslabs%2Fmcp%20user_agent&type=code

`f"awslabs/mcp/healthimaging-mcp-server/{__version__}"`